internal/datastore/filetracker/filetracker.go

```diff
@@ -37,16 +42,26 @@ func New(r Repository) *FileTracker {
 	return &FileTracker{
 		repo:   r,
 		Hasher: xxHash32Hasher{},
+		logger: log.GetInstance(),
```

Comment:
This logger could be different from the app.Logger. With the current implementation it's going to be the same, but the logger may should be injected.

This is not something you should address, but it's one of the problems I saw to use loggers inside some parts of the code :-D